portfolio/src/main/webapp/script.js

```diff
@@ -77,15 +77,25 @@ function fetchComments(quantity=5) {
       document.getElementById('delete-comments-button').setAttribute('disabled','true');
       commentContent = '<div class="comment"><p class="body-text">No Comments</p></div>';
     }
-    else { // otherwise display the comments
+    else { // otherwise display the comments and any attached image
       for (let i = 0; i < commentData.length; i++) {
         let commentText = commentData[i].commentText; // The actual comment
         
         // Comment author name: "'Anonymous' if there was not a name submitted."
         let commentAuthor = commentData[i].commentAuthor === "" ? "Anonymous" : commentData[i].commentAuthor; 
-
-        commentContent += '<div class="comment"><p class="body-text"><b>' + commentAuthor + '</b></p>' 
-            + '<p class="body-text">' + commentText + '</p></div>';
+        
+        // Display image as well, if there is an image
+        let imageUrl = commentData[i].attachedImage;
+        if (imageUrl == null) {
```

Comment:
|commentContent| share the same first part (|commentAuthor| and |commentText|) with or without |imageUrl|. Could you take that into account and make this part cleaner?

Reply:
I have two different solutions for this one - which one is better practice do you think?

I could do it like this, just moving the shared section of the string outside of the if-block:

```
// Display image as well, if there is an image
let imageUrl = commentData[i].attachedImage;
commentContent += '<div class="comment"><div class="flex-item"><p class="body-text"><b>' + commentAuthor + '</b></p>' 
    + '<p class="body-text">' + commentText + '</p></div>'; // outer <div> not closed yet

if (imageUrl == null) {
  commentContent += '</div>'; // close outer div
}
else {
  commentContent += '<div class="flex-item"><a href=' + imageUrl + ' target="_blank"><img class="comment-image" src=' + imageUrl + '></a></div></div>'; // add image
}
```

or I could do it like this, using `document.createElement` to generate all the elements instead of writing HTML in strings:

There is a lot more lines of code, but it might make it easier to extend the comments display later on.

```
function fetchComments(quantity=5) {
  
  const endpoint = '/data?';
  var queryString = new URLSearchParams();
  queryString.append('quantity', String(quantity));
  const url = endpoint + queryString.toString();

  fetch(url).then(response => response.json()).then((commentData) => {
    
    // Delete what is currently displayed
    document.getElementById('comment-list').innerHTML = "";
    
    if (commentData.length === 0 ) { // if there are no existing comments
      var commentDiv = document.createElement('div'); // div to store one comment
      commentDiv.className = "comment";
      
      // disable the delete comments button & display "No Comments"
      document.getElementById('delete-comments-button').setAttribute('disabled','true');
      let content = document.createElement('p');
      content.className = "body-text";
      content.innerText = "No Comments";
      commentDiv.appendChild(content);

      document.getElementById('comment-list').appendChild(commentDiv);

    }
    else { // otherwise display the comments and any attached image
      
      for (let i = 0; i < commentData.length; i++) {
        
        var commentDiv = document.createElement('div'); // div to store one comment
        commentDiv.className = "comment";

        let commentText = commentData[i].commentText; 
        // Comment author name: "'Anonymous' if there was not a name submitted."
        let commentAuthor = commentData[i].commentAuthor === "" ? "Anonymous" : commentData[i].commentAuthor; 
        let imageUrl = commentData[i].attachedImage;

        let textContainer = document.createElement('div'); // Container for comment text & author name
        textContainer.className = "flex-item";
        
        // Author name <p> element
        let authorP = document.createElement('p');
        authorP.className = "body-text";
        let bolding = document.createElement('b');
        bolding.innerText = commentAuthor;
        authorP.appendChild(bolding);

        // Comment text <p> element
        let commentP = document.createElement('p');
        commentP.className = "body-text";
        commentP.innerText = commentText;

        textContainer.appendChild(authorP);
        textContainer.appendChild(commentP);
        commentDiv.appendChild(textContainer);

        // Display image as well, if there is an image
        if (imageUrl) {
          let imageContainer = document.createElement('div');
          imageContainer.className = "flex-item";
          
          let imageLink = document.createElement('a');
          imageLink.href = imageUrl;
          imageLink.target = "_blank"; // link will open in new tab
          
          let image = document.createElement('img');
          image.className = "comment-image";
          image.src = imageUrl;

          imageLink.appendChild(image);
          imageContainer.appendChild(imageLink);
          commentDiv.appendChild(imageContainer);
        }
        document.getElementById('comment-list').appendChild(commentDiv);
      }
    }
  });
}
```

Reply:
I would go with the 1st option. 2nd option is nice, but is a bit overkill imo.